core/src/main/java/com/novoda/noplayer/drm/DrmSessionCreatorFactory.java

```diff
@@ -18,7 +18,7 @@ public DrmSessionCreator createFor(DrmType drmType, DrmHandler drmHandler) {
                 );
                 return StreamingDrmSessionCreator.newInstance(mediaDrmCallback, new FrameworkMediaDrmCreator());
             case WIDEVINE_MODULAR_DOWNLOAD:
-                return new NoDrmSessionCreator();
+                return new DownloadDrmSessionCreator((DownloadedModularDrm) drmHandler, new FrameworkMediaDrmCreator());
```

Comment:
`CreatorFactory` is a bit of a `FactoryFactory` type thing isn't it? Can we not just make a class that returns the correct `DrmSessionManager` implementation directly?